Makefile

```diff
@@ -10,13 +10,33 @@ DARKGRAY=\033[1;30m
 YELLOW=\033[0;33m
 NC=\033[0m # No Color
 
-TEST_LEADER = none
-TEST_MOTOR = none
-TEST_CAR = none
-
-DEBUG = 0
+DEBUG ?= 1
 export DEBUG
 
+# Note: ?= will define the variable only if it is not already set, which means the user can override any of
+# the TEST_LEADER, TEST_MOTOR, or TEST_CAR assignments with their own test files.
+
+# Check if test file exists for the leader.
+ifneq (,$(wildcard Tests/Leader/Test_$(TEST).c))
+	TEST_LEADER ?= Tests/Leader/Test_$(TEST).c
+else
+	TEST_LEADER ?= Apps/Src/main.c
```

Comment:
Shouldn't we just give them a warning

Reply:
Nah, it just means the default is using main instead of a test.